endpoint/labels.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoint
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+var (
+	// ErrInvalidHeritage is returned when heritage was not found, or different heritage is found
+	ErrInvalidHeritage = errors.New("heritage is unknown or not found")
+)
+
+const (
+	heritage = "external-dns"
+	// OwnerLabelKey is the name of the label that defines the owner of an Endpoint.
+	OwnerLabelKey = "owner"
+	// ResourceLabelKey is the name of the label that identifies k8s resource which wants to acquire the DNS name
+	ResourceLabelKey = "resource"
```

Comment:
`Owner` was supposed to identify the owner of a resource. For us it was to say it's "the cluster" (or rather the - usually one - instance of external-dns). However now the owner is a particular resource in a particular namespace in a particular cluster. I wonder if it's confusing to separate them this way.

I would rather clarify it with either a combined owner resource (`owner=extdns1/default/my-service`, is that even easier to code?) or two owner related labels (`ownerInstance=extdns1, ownerResource=default/my-service`, just a rename then). 

Although there's the concern of backwards compatibility...

@ideahitme let me know what you think.

Reply:
yeah I agree, the naming is probably not the most descriptive. 

> However now the owner is a particular resource in a particular namespace in a particular cluster. I wonder if it's confusing to separate them this way.

I would still prefer them to be separated:
- "owner" label signifies if the DNS Record can be modified at all
- "resource" label helps to figure out how it should be updated

I also can see scenarios where "resource" label can be ignored altogether, so in mind it is just a metadata, which allows to implement a "method" for determining resource-ownership, but it could be used for something else, or a different method could be implemented. That's why I thought it is better to keep the label name independent of how we use it.

Reply:
Agreed!